package.json

```diff
@@ -88,5 +89,9 @@
     "typescript": "4.3.5",
     "uuid": "8.3.2",
     "webpack": "5.38.1"
+  },
+  "resolutions": {
+    "@types/graphql": "14.5.0",
+    "graphql": "15.5.1"
   }
```

Comment:
No it's in package.json ([see here](https://classic.yarnpkg.com/en/docs/selective-version-resolutions))

Reply:
The link doesn't really tells that it's not doable from `.yarnrc.yml`. But I've found the information [here](https://yarnpkg.com/configuration/yarnrc#packageExtensions), you're right, it's only usable in the `package.json` file.

As I now understand better what's the purpose of this option and the difference with `packageExensions`, I think that it could be avoidable by pinning exact versions in the `package.json` dependencies instead of prefixing them with `^` ([doc](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies)).

Reply:
I do have exact versions of `graphql` package. However, `apollo` package comes with a `graphql` dependency which is probably not the same than ours.

So if you remove the `resolutions` block, you will get the following error:

```
Error: Command failed: yarn codegen:all
    Error: Cannot use GraphQLScalarType "ID" from another module or realm.

    Ensure that there is only one instance of "graphql" in the node_modules
    directory. If different versions of "graphql" are the dependencies of
    other
    relied on modules, use "resolutions" to ensure only one version is
    installed.

    https://yarnpkg.com/en/docs/selective-version-resolutions

    Duplicate "graphql" modules cannot be used at the same time since
    different
    versions may have different capabilities and behavior. The data from one
    version used in the function from another could produce confusing and
    spurious results.
```

PS: we can remove the line about graphql types